runtime/ts/entity.ts

```diff
@@ -5,15 +5,20 @@
 // Code distributed by Google as part of this project is also
 // subject to an additional IP rights grant found at
 // http://polymer.github.io/PATENTS.txt
-'use strict';
 
-import {assert} from '../platform/assert-web.js';
-import {Symbols} from './ts-build/symbols.js';
-import {Type} from './ts-build/type.js';
+import {assert} from '../../platform/assert-web.js';
+import {Symbols} from './symbols.js';
+import {Type} from './type.js';
+import {Schema} from './schema.js';
 
-export class Entity {
-  constructor(userIDComponent) {
-    assert(!userIDComponent || userIDComponent.indexOf(':') == -1, 'user IDs must not contain the \':\' character');
+export abstract class Entity {
+  private _userIDComponent: string|null;
+  // TODO determine the right type to use for the proxy in schema.ts
+  // tslint:disable-next-line: no-any
+  protected rawData: any;
+
+  constructor(userIDComponent: string|null) {
```

Comment:
provide with a default (== null), or change null to void or undefined to capture non-provided userIDComponent case.

Reply:
done.